### PR TITLE
signin page for beta environment

### PR DESCRIPTION
**Pull request by Ayham1590**
sign in and sign up page UI

@chinathaip 
sign in button on header, navbar, and session modal will now redirect to the signing page as long as the APP_ENV is not `production`.

### CODE REVIEW

File: apps/web/app/signin/page.tsx
Comment (CryoMilo) [inline apps/web/app/signin/page.tsx:1]: Hi bro @Ayham1590 

- We would need a signup page and a link to access to it. 
- We would need a Login button too.

Screenshot 2024-03-13 at 11 01 52 PM

**Review by CryoMilo — CHANGES_REQUESTED** [review]

**Comment by chinathaip** [comment]
FYI, this issue closes https://github.com/stamford-syntax-club/course-compose/issues/49

File: apps/web/app/signin/page.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    Button,
+    Divider,
+    Checkbox,
+    Anchor,
+    Stack,
+} from '@mantine/core';
+
+export default function SigninPage(props: PaperProps) {
+    const [type, toggle] = useToggle(['login', 'register']);
+    const [showRepeatPassword, setShowRepeatPassword] = useState(false);
```
Comment (chinathaip) [inline apps/web/app/signin/page.tsx:22]: is this state being referred to anywhere? If not, let's remove it

File: apps/web/app/signin/page.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
+"use client";
+
+import { useState } from 'react'; // Import useState hook
+import { useToggle, upperFirst } from '@mantine/hooks';
```
Comment (chinathaip) [inline apps/web/app/signin/page.tsx:4]: is `upperFirst` being used to anywhere? If not, let's remove it

File: apps/web/app/signin/page.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    Group,
+    PaperProps,
+    Button,
+    Divider,
+    Checkbox,
+    Anchor,
+    Stack,
+} from '@mantine/core';
+
+export default function SigninPage(props: PaperProps) {
```
Comment (chinathaip) [inline apps/web/app/signin/page.tsx:20]: I'm a bit confused with this :(

Can you explain why are we passing paper props to this component?

React actually complains about this too

image

File: apps/web/app/signin/page.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    const [type, toggle] = useToggle(['login', 'register']);
+    const [showRepeatPassword, setShowRepeatPassword] = useState(false);
+    const form = useForm({
+        initialValues: {
+            email: '',
+            name: '',
+            password: '',
+            repeatPassword: '',
+            terms: true,
```
Comment (Lxkas) [inline apps/web/app/signin/page.tsx:29]: Are you sure this should default to true?

**Review by Lxkas — CHANGES_REQUESTED** [review]

**Review by Lxkas — APPROVED** [review]
LGTM!

**Review by NayHtetKyaw — APPROVED** [review]
GUDGUD

**Pull request merged**